noir-projects/aztec-nr/aztec/src/discovery/private_logs.nr

```diff
@@ -134,8 +136,8 @@ unconstrained fn decode_log_plaintext(
     // See the documentation of this function for a description of the log layout
     let expanded_log_metadata = log_plaintext.get(0);
 
-    let log_type_id = ((expanded_log_metadata as u64) / 128);
-    let log_metadata = ((expanded_log_metadata as u64) % 128);
+    let log_metadata = ((expanded_log_metadata as u128) >> 64) as u64;
+    let log_type_id = (expanded_log_metadata as u64);
```

Comment:
Casting and shifting in Noir has always been troublesome. Given we'll continue to update this, extract this snippet into a fn that extracts both u64 from the u128, and let's add a couple tests for it including the edge cases (e.g. 0xffff, 0xffff00000, 0x0000fffff, etc)

Reply:
Good idea. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/12956/commits/bde6715045a08cabfdf77cfb743d5f1c9daee32c